src/solver/batchnorm/forward_inference_ck.cpp

```diff
@@ -126,52 +126,60 @@ template <typename XDataType,
           typename ScaleDataType,
           typename BiasDataType,
           typename MeanVarDataType>
-static void RunCKSolution(const Handle& handle,
-                          const AnyInvokeParams& primitive_parameters,
-                          const miopen::batchnorm::ProblemDescription& problem)
+ConvSolution InvokerFactoryMakerNHWC(const miopen::batchnorm::ProblemDescription& bn_problem)
 {
-    const auto& args = CKArgsBNormFwd{problem};
-
-    using DeviceOp = ck::tensor_operation::device::DeviceElementwise<
-        ck::Tuple<XDataType, MeanVarDataType, MeanVarDataType, ScaleDataType, BiasDataType>,
-        ck::Tuple<YDataType>,
-        Normalize,
-        Rank>;
-    const auto bn_fwd_ptrs = ck::tensor_operation::device::instance::DeviceOperationInstanceFactory<
-        DeviceOp>::GetInstances();
+    ConvSolution result;
+    result.invoker_factory = [args         = CKArgsBNormFwd{bn_problem},
+                              kernel_index = CheckCKApplicability<XDataType,
+                                                                  YDataType,
+                                                                  AccDataType,
+                                                                  ScaleDataType,
+                                                                  BiasDataType,
+                                                                  MeanVarDataType>(bn_problem)](
+                                 const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [&](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
```

Comment:
@atamazov @DrizztDoUrden can we guarantee that `invoker_factory` context exists longer that its generated lambda?
I'm asking about this case:
```C++
result.invoker_factory = [=](...)
{
    retunr [&](...){...};
};
```
versus that one:
```C++
result.invoker_factory = [=](...)
{
    retunr [=](...){...};
};
```

If we can guarantee that, we can always capture by reference in the second lambda and save some memory and avoid extra copy operations. If we can't, we must always capture by value.

Reply:
No, it **almost always** exists for a shorter period. Capturing by a reference here is an error, both lambdas should always only copy, never referencing.

Reply:
@CAHEK7 see 
- #216